bitsail-connectors/connector-hbase/src/main/java/com/bytedance/bitsail/connector/hbase/HBaseHelper.java

```diff
@@ -0,0 +1,133 @@
+
+  /**
+   * Get connection with hbase.
+   * @param hbaseConfigMap HBase configuration.
+   */
+  public static Connection getHbaseConnection(Map<String, Object> hbaseConfigMap) {
+    Preconditions.checkState(MapUtils.isNotEmpty(hbaseConfigMap), "HBaseConfig cannot be empty!");
+
+    if (KerberosAuthenticator.openKerberos(hbaseConfigMap)) {
+      return getConnectionWithKerberos(hbaseConfigMap);
+    }
+
+    try {
+      Configuration hbaseConfiguration = getConfig(hbaseConfigMap);
+      return ConnectionFactory.createConnection(hbaseConfiguration);
+    } catch (IOException e) {
+      LOG.error("Get connection fail with config:{}", hbaseConfigMap);
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Get hadoop configuration for hbase.
+   * @param hbaseConfigMap User defined hbase configurations.
+   * @return A hadoop configuration.
+   */
+  public static Configuration getConfig(Map<String, Object> hbaseConfigMap) {
+    Configuration hbaseConfiguration = HBaseConfiguration.create();
+    if (MapUtils.isEmpty(hbaseConfigMap)) {
+      return hbaseConfiguration;
+    }
+
+    for (Map.Entry<String, Object> entry : hbaseConfigMap.entrySet()) {
+      if (entry.getValue() != null && !(entry.getValue() instanceof Map)) {
+        hbaseConfiguration.set(entry.getKey(), entry.getValue().toString());
+      }
+    }
+    return hbaseConfiguration;
+  }
+
+  /**
+   * Close mutator.
+   */
+  public static void closeBufferedMutator(BufferedMutator bufferedMutator) {
+    closeWithException(bufferedMutator);
+  }
+
+  /**
+   * Close hbase connection.
+   */
+  public static void closeConnection(Connection hbaseConnection) {
+    closeWithException(hbaseConnection);
+  }
+
+  /**
+   * Login kerberos based on configurations, and then get connection with hbase.
+   * @param hbaseConfigMap HBase configuration.
+   * @return Connection with hbase.
+   */
+  private static Connection getConnectionWithKerberos(Map<String, Object> hbaseConfigMap) {
+    try {
+      UserGroupInformation ugi = KerberosAuthenticator.getUgi(hbaseConfigMap);
+      return ugi.doAs((PrivilegedAction<Connection>) () -> {
+        try {
+          Configuration hbaseConfiguration = getConfig(hbaseConfigMap);
+          return ConnectionFactory.createConnection(hbaseConfiguration);
+        } catch (IOException e) {
+          LOG.error("Get connection fail with config:{}", hbaseConfigMap);
+          throw new RuntimeException(e);
+        }
+      });
+    } catch (Exception e) {
+      throw new RuntimeException("Login kerberos error", e);
+    }
+  }
+
+  /**
+   * Close an object and throw exception if failed to close.
+   */
+  private static void closeWithException(Closeable closeable) {
+    try {
+      if (Objects.nonNull(closeable)) {
+        closeable.close();
+      }
+    } catch (Exception e) {
+      throw new RuntimeException("Failed to close.", e);
+    }
+  }
+
+  @Override
+  public void close() throws Exception {
```

Comment:
It looks like `AutoCloseable` interface is not necessary.

Reply:
Removed AutoCloseable as suggested.